snuba/utils/streams/consumers/consumer.py

```diff
@@ -0,0 +1,122 @@
+from typing import Callable, Generic, Mapping, Optional, Sequence
+
+from snuba.utils.streams.consumers.backends.abstract import ConsumerBackend
+from snuba.utils.streams.consumers.types import Message, TStream, TOffset, TValue
+
+
+class Consumer(Generic[TStream, TOffset, TValue]):
+    """
+    This class provides an interface for consuming messages from a
+    multiplexed collection of streams. The specific implementation of how
+    messages are consumed is delegated to the backend implementation.
+
+    This interface is heavily "inspired" by the Confluent Kafka Consumer
+    implementation, but only exposes a limited set of the available methods
+    and several method signatures and/or return values differ -- these
+    changes are intended to improve the ease of use of the consumer, as well
+    as make development and testing more straightforward. Most code that uses
+    the Confluent driver can be modified to use this interface with minimal
+    changes, but it is not intended to be a drop-in replacement for the
+    Confluent implementation.
```

Comment:
This comment is present in the ConsumerBackend class.  Is it actually relevant here ?

Reply:
It's somewhat relevant but becoming less and less true and more vestigial. I think it's fine to remove it at this point to remove any implied association with the Kafka consumer.

It should probably be removed from `ConsumerBackend` as well. Since that class should not be used directly, the user shouldn't be concerned with where the API drew inspiration from at one point.